developer-extra.yml

```diff
@@ -14,6 +14,7 @@ channels:
   - conda-forge
 
 dependencies:
+  - nomkl  # mkl leads to large conda env, slowing down CI and env caching
```

Comment:
What are we using it for? Can this cause problems downstream with, e.g., Mantid? A release build probably has to include MKL.

Reply:
This is used by numpy and friends. I deliberately did *not* change this on `conda/meta.yml`, i.e., this would affect only CI builds.

Reply:
And potentially developer's environments

Reply:
I think developers should use a different env file. I also removed Sphinx and all tools like flake8, so these envs are not very useful.

Reply:
Devs could continue using the script to generate an env.

Note the I'd remove the nomkl here if we keep the files for the build matrix (below). This is actually just a leftover from an intermediate attempt.

Reply:
> I think developers should use a different env file. I also removed Sphinx and all tools like flake8, so these envs are not very useful.

Then we should move this file (or platform specific ones) out of the root directory.

Reply:
I think `libblas=*=*openblas` is preferred over `nomkl` for some reason I don't remember

Reply:
Thanks, I'll have a look.